### ISSUE TITLE
Make a more flexible background model for the inter-order scattered light / structured dark current

### ISSUE THREAD

**Issue by gully** [gully/ynot#9]
We currently assume a constant scalar background structure. The inter-order residuals are now among the most conspicuous features when training is allowed to proceed indefinitely.  Hooray, that's a good problem to have since the out-of-slit background is the least important nuisance feature remaining!  In fact it's so unimportant we can almost just ignore it.  

But if we wanted to, we could fit a more flexible model like a polynomial surface in `x` and `y`, or we could consider a GP model for the inter-order scattered light. The inter-order background mostly just affects the zero-point if we were trying to determine what the continuum sky background emission is---mostly of academic interest---or it would become important if we were observing extended objects that fill the slit, like a solar system planet like Jupiter.  Training a GP is tricky because we would have two tasks: An out-of-slit GP model and an in-slit GP model.  That's fine, we just add the losses, and backpropagate through both GPs, but in practice sounds tiresome, and a bit of overkill.

The problem with a more flexible model like a polynomial surface in `x` and `y` is that it can become too flexible and over-fit genuine structure in the sky or along a cluster of bright sky lines.  So we almost want to mask the sky traces regardless.  Another option could be to pre-process the background by masking all of the echelle traces.  That's not a bad idea.
